src/ResourceManager/Network/Commands.Network/NetworkWatcher/ConnectionMonitor/SetAzureNetworkWatcherConnectionMonitorCommand.cs

```diff
@@ -0,0 +1,262 @@
+            {
+                resourceGroupName = this.ResourceGroupName;
+                networkWatcherName = this.NetworkWatcherName;
+            }
+            else if (ParameterSetName.Contains("SetByInputObject"))
+            {
+                ConnectionMonitorDetails connectionMonitorDetails = new ConnectionMonitorDetails();
+                connectionMonitorDetails = this.GetConnectionMonitorDetails(this.InputObject.Id);
+
+                connectionMonitorName = connectionMonitorDetails.ConnectionMonitorName;
+                resourceGroupName = connectionMonitorDetails.ResourceGroupName;
+                networkWatcherName = connectionMonitorDetails.NetworkWatcherName;
+            }
+            else
+            {
+                var networkWatcher = this.GetNetworkWatcherByLocation(this.Location);
+
+                if (networkWatcher == null)
+                {
+                    throw new ArgumentException("There is no network watcher in the specified location");
+                }
+
+                resourceGroupName = NetworkBaseCmdlet.GetResourceGroup(networkWatcher.Id);
+                networkWatcherName = networkWatcher.Name;
+            }
+
+            var present = this.IsConnectionMonitorPresent(resourceGroupName, networkWatcherName, connectionMonitorName);
+
+            if (!present)
+            {
+                throw new ArgumentException(Microsoft.Azure.Commands.Network.Properties.Resources.ResourceNotFound);
+            }
+
+            var connectionMonitor = UpdateConnectionMonitor();
+            WriteObject(connectionMonitor);
+        }
+
+        private PSConnectionMonitorResult UpdateConnectionMonitor()
+        {
+            MNM.ConnectionMonitor parameters = new MNM.ConnectionMonitor
+            {
+                Source = new MNM.ConnectionMonitorSource
+                {
+                    ResourceId = this.SourceResourceId,
+                    Port = this.SourcePort
+                },
+                Destination = new MNM.ConnectionMonitorDestination
+                {
+                    ResourceId = this.DestinationResourceId,
+                    Address = this.DestinationAddress,
+                    Port = this.DestinationPort
+                },
+                Tags = TagsConversionHelper.CreateTagDictionary(this.Tag, validate: true)
+            };
+
+            if (this.AutoStart != null)
+            {
+                parameters.AutoStart = this.AutoStart;
+            }
+
+            if (this.MonitoringIntervalInSeconds != null)
+            {
+                parameters.MonitoringIntervalInSeconds = this.MonitoringIntervalInSeconds;
+            }
+
+            PSConnectionMonitorResult getConnectionMonitor = new PSConnectionMonitorResult();
+
+            // Execute the CreateOrUpdate Connection monitor call
+            if (ParameterSetName.Contains("SetByResource") && !ParameterSetName.Contains("SetByResource"))
```

Comment:
@irrogozh this will never happen - I think this is a typo ☺️

Reply:
Fixed:) Thanks!